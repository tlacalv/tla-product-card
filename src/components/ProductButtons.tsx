import styles from "../styles/styles.module.css";
import React, { CSSProperties, useCallback, useContext } from "react";
import { ProductContext } from "./ProductCard";

export interface Props {
  className?: string;
  style?: CSSProperties;
}

export const ProductButtons = ({ className, style }: Props) => {
  const { increaseBy, counter, maxCount } = useContext(ProductContext);

  const isMaxReached = useCallback(
    () => !!maxCount && maxCount===counter,
    [counter, maxCount]
  )

  return (
    <div className={`${styles.buttonsContainer} ${className}`} style={style}>
      <button className={styles.buttonMinus} onClick={() => increaseBy(-1)}>
        -
      </button>
      <div className={styles.countLabel}>{counter}</div>
      <button disabled={isMaxReached()} className={`${styles.buttonAdd} ${isMaxReached() && styles.disabled}`} onClick={() => increaseBy(1)}>
        +
      </button>
    </div>
  );
};
