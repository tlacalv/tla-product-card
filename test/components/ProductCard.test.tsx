import React from 'react'
import renderer from 'react-test-renderer'
import { ProductCard} from '../../src/components';
import { product1 } from '../data/products';
const {act} = renderer;

describe('ProductTitle', () => {
  test('must show the component without errors', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          () => (
            <h1>Product card</h1>
          )
        }
      </ProductCard>
    )

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
  test('must increase counter', () => {
    const wrapper = renderer.create(
      <ProductCard product={product1}>
        {
          ({count, increaseBy}) => (
            <>
              <h1>Product card</h1>
              <span>{count}</span>
              <button onClick={()=>increaseBy(1)}></button>
            </>
          )
        }
      </ProductCard>
    )
    let tree = wrapper.toJSON();
    expect(tree).toMatchSnapshot();

    act(()=> {
      (tree as any).children[2].props.onClick();
    })

    tree = wrapper.toJSON();

    expect( (tree as any).children[1].children[0]).toBe('1')
  });

})