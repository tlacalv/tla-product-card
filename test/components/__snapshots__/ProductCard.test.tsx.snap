// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle must increase counter 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
  <span>
    0
  </span>
  <button
    onClick={[Function]}
  />
</div>
`;

exports[`ProductTitle must show the component without errors 1`] = `
<div
  className="productCard undefined"
>
  <h1>
    Product card
  </h1>
</div>
`;
